controllers/kubeadmbootstrapconfig_controller.go

```diff
@@ -23,12 +23,61 @@ import (
 	"github.com/go-logr/logr"
 	"github.com/pkg/errors"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	kubeadmv1alpha1 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1"
-	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha2"
+	capiv1alpha2 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+// Add creates a new KubeadmBootstrap Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) *KubeadmBootstrapConfigReconciler {
+	return &KubeadmBootstrapConfigReconciler{
+		Client: mgr.GetClient(),
+		Log:    nil, //TODO wait for main.go implementation to pass down logger
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("kubeadm-bootstrap-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to Cluster API Machine
+	err = c.Watch(
+		&source.Kind{Type: &capiv1alpha2.Machine{}},
+		&handler.EnqueueRequestForObject{},
+	)
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(
+		&source.Kind{Type: &capiv1alpha2.Machine{}},
+		&handler.EnqueueRequestsFromMapFunc{
```

Comment:
This should be moved to in SetupWithManager using https://github.com/kubernetes-sigs/controller-runtime/blob/bace65823215cf3226bd0395fab06f0fe798c935/pkg/builder/controller.go#L99